py/desispec/scripts/trace_shifts.py

```diff
@@ -7,7 +7,7 @@
 import os, sys
 import argparse
 import numpy as np
-from numpy.linalg.linalg import LinAlgError
+from numpy.linalg import LinAlgError
```

Comment:
Stylistic question here: in `fluxcalibration.py`, the exception catches `numpy.linalg.LinAlgError` and presumably in this file it's just `catch LinAlgError`. Both work, but do we want to prefer one style over the other?

Reply:
See also `sky.py`.

Reply:
For custom exceptions that aren't part of the Python standard namespace, I personally prefer using the longer fully qualified path `np.linalg.LinAlgError` at the point of catching the exception to make it more obvious where it came from without having to scroll back to the top to read the imports. But I don't mind the other form either. For the purposes of this PR, I tried to make only the minimal changes needed to support numpy 2 without delving into stylistic cleanup that wasn't strictly necessary.